hoomd/md/EvaluatorPairSMie.h

```diff
@@ -0,0 +1,211 @@
+#include <string>
+#endif
+
+#include "hoomd/HOOMDMath.h"
+#include "EvaluatorPairLJ.h"
+
+/*! \file EvaluatorPairSMie.h
+    \brief Defines the pair evaluator class for shifted Mie potentials
+*/
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+// DEVICE is __host__ __device__ when included in nvcc and blank when included into the host compiler
+#ifdef __HIPCC__
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+//! Class for evaluating the Shifter Mie pair potential
+/*! <b>General Overview</b>
+
+    See EvaluatorPairSLJ and EvaluatorPairMie
+
+    <b>SMie specifics</b>
+
+    EvaluatorPairSMie evaluates the function:
+    \f{eqnarray*}
+    V_{\mathrm{SMie}}(r)  = \left( \frac{n}{n-m} \right) {\left( \frac{n}{m} \right)}^{\frac{m}{n-m}} \varepsilon \left[ \left( \frac{\sigma}{r-\Delta} \right)^{n} -
+                \left( \frac{\sigma}{r-\Delta} \right)^{m} \right] r \l (r_{\mathrm{cut}} + \Delta)\\
+                = & 0 & r \ge (r_{\mathrm{cut}} + \Delta) \\
+    \f}
+    where  \f$ \Delta is specified by the user, conventionally calcualted as, \f$ \Delta = (d_i + d_j)/2 - \sigma \f$ and \f$ d_i \f$ is the diameter of particle \f$ i \f$.
+
+    The SMie potential does not need charge, but does need diameter. Two parameters are specified and stored in a
+    Scalar5. \a sm1 \a sm2 \a sm3 \a sm4 \a sm5 are stored in \a params.x \a params.y \a params.z and \a params.w \a params.v respectively.
+
+
+    These are related to the standard lj parameters sigma and epsilon and the variable exponents n and m by:
+    - \a sm1 = epsilon * pow(sigma,n) * (n/(n-m)) * power(n/m,m/(n-m))
+    - \a sm2 = epsilon * pow(sigma,m) * (n/(n-m)) * power(n/m,m/(n-m))
+    - \a sm3 = n
+    - \a sm4 = m
+    - \a sm5 = Delta
+
+    Due to the way that SMie modifies the cutoff condition, it will not function properly with the xplor shifting mode.
+*/
+class EvaluatorPairSMie
+    {
+    public:
+        //! Define the parameter type used by this pair potential evaluator
+        struct param_type
+            {
+            Scalar sm1;
+            Scalar sm2;
+            Scalar sm3;
+            Scalar sm4;
+            Scalar sm5;
+
+            #ifndef ENABLE_HIP
+            //! set CUDA memory hints
+            void set_memory_hint() const {}
+            #endif
+
+            #ifndef __HIPCC__
+            param_type() : sm1(0), sm2(0), sm3(0), sm4(0), sm5(0) {}
+
+            param_type(pybind11::dict v)
+                {
+                sm3 = v["n"].cast<Scalar>();
+                sm4 = v["m"].cast<Scalar>();
+
+                auto sigma(v["sigma"].cast<Scalar>());
+                auto epsilon(v["epsilon"].cast<Scalar>());
+
+                Scalar outFront = (sm3/(sm3-sm4)) * fast::pow(sm3/sm4, sm4/(sm3-sm4));
+                sm1 = outFront * epsilon * fast::pow(sigma, sm3);
+                sm2 = outFront * epsilon * fast::pow(sigma, sm4);
+
+                sm5 = v["Delta"].cast<Scalar>();
```

Comment:
```suggestion
                delta = v["delta"].cast<Scalar>();
```
You are right that this is a capital delta from a language perspective, from a code style perspective users will expect the parameter dictionary keys to be named like variables which are all lowercase with underscores separating words. There is no lowercase delta to confuse this with in the potential, so it is not a problem.

Also, feel free to name struct members with more meaningful names than `smN`.